tests/samples/matmul_fill_static.mlir

```diff
@@ -1,9 +1,22 @@
-func.func @matmul_static(%lhs : tensor<128x256xf32>,
-    %rhs : tensor<256x512xf32>) -> tensor<128x512xf32> {
-  %empty = tensor.empty() : tensor<128x512xf32>
-  %cst = arith.constant 0.0 : f32
-  %fill = linalg.fill ins(%cst : f32) outs(%empty : tensor<128x512xf32>) -> tensor<128x512xf32>
-  %2 = linalg.matmul ins(%lhs, %rhs : tensor<128x256xf32>, tensor<256x512xf32>)
-      outs(%fill : tensor<128x512xf32>) -> tensor<128x512xf32>
-  return %2 : tensor<128x512xf32>
+// Example using a transform dialect script.
+
+// Compile to mlir-aie:
+
+// RUN: iree-compile --iree-hal-target-backends=amd-aie --compile-to=executable-sources %s | iree-opt --pass-pipeline="builtin.module(hal.executable(hal.executable.variant(iree-hal-translate-target-executable-variants{target=amd-aie})))" --iree-codegen-transform-dialect-library=%S/matmul_fill_spec_pad.mlir | FileCheck %s
+
+// To compile to just after the transform dialect pass: // RUN: iree-opt %s   --iree-hal-target-backends=amd-aie --iree-abi-transformation-pipeline --iree-flow-transformation-pipeline --iree-stream-transformation-pipeline --iree-hal-configuration-pipeline | cat - %S/matmul_fill_spec_pad.mlir | iree-opt --iree-transform-dialect-interpreter | FileCheck %s
```

Comment:
These two `RUN` commands need to combined into one. The first `iree-opt` in this `RUN` command and the first `iree-compile` in the first `RUN` command are essentially the same.

For the second command to run the transform dialect script maybe follow what is here https://github.com/openxla/iree/blob/eaefd3ff964f14e700b1c468aa2b4f20a4401028/tests/transform_dialect/cpu/matmul.mlir#L18 (ignore the first `iree-opt` command, it is doing what you are doing here when it doesnt need to.. bonus points for fixing that as well :P )

Reply:
I updated it, but I'm not 100% sure this is what you have in mind...